src/scrabble.js

```diff
@@ -1,38 +1,124 @@
-// function scrabble() {}
-// module.exports = scrabble
-
 const alphaGroup = [
   ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T', 1],
   ['D', 'G', 2],
   ['B', 'C', 'M', 'P', 3],
   ['F', 'H', 'V', 'W', 'Y', 4],
   ['K', 5],
   ['J', 'X', 8],
-  ['Q', 'Z', 10],
-  ['{', '}', 2],
-  ['[', ']', 3]
+  ['Q', 'Z', 10]
 ]
 
-function scrabble(word) {
-  // define a variable to store the scrabble result
+const scrabble = (word) => {
   let scrabbleScore = 0
-  // test for null to return 0
   if (word === null) return 0
-  // convert the scrabble word to uppercase and then split it
   const scrabbleWord = word.toUpperCase().split('')
-  // the outer for loop will iterate through the scrabble word elements
+  let doubleTheWord = []
+  let tripleTheWord = []
+
+  // d{og
+  if (scrabbleWord.includes('{')) {
+    if (scrabbleWord.indexOf('}') === -1) {
+      return scrabbleScore
+    }
+  }
+
+  if (scrabbleWord.includes('}')) {
+    if (scrabbleWord.indexOf('{') === -1) {
+      return scrabbleScore
+    }
+  }
+
+  if (scrabbleWord.includes('[')) {
+    if (scrabbleWord.indexOf(']') === -1) {
+      return scrabbleScore
+    }
+  }
+
+  if (scrabbleWord.includes(']')) {
+    if (scrabbleWord.indexOf('[') === -1) {
+      return scrabbleScore
+    }
+  }
+
+  if (scrabbleWord.includes('{') && scrabbleWord.includes('}')) {
+    const startIndex = scrabbleWord.indexOf('{')
+    const endIndex = scrabbleWord.indexOf('}')
+    doubleTheWord = scrabbleWord.splice(
+      startIndex + 1,
+      endIndex - (startIndex + 1)
+    )
+
+    for (let i = 0; i < doubleTheWord.length; i++) {
+      for (let j = 0; j < alphaGroup.length; j++) {
+        if (alphaGroup[j].includes(doubleTheWord[i])) {
+          scrabbleScore += alphaGroup[j][alphaGroup[j].length - 1] * 2
+        }
+      }
+    }
+  }
+
+  if (scrabbleWord.includes('[') && scrabbleWord.includes(']')) {
```

Comment:
I like the tidiness of checking that the [ ] have been properly interred in order to apply the multipliers!